### ISSUE TITLE
[method-signature-style] Autofixer generates uncompilable code

### ISSUE THREAD

**Issue by merlinnot** [typescript-eslint/typescript-eslint#1911]
**Repro**

Configuration:

```JSON
{
  "rules": {
    "@typescript-eslint/method-signature-style": "error"
  }
}
```

Source code:

```typescript
interface MyEventEmitter extends EventEmitter {
  on: (event: 'my-event-1', listener: () => void | Promise<void>) => this;
  on: (
    event: 'error',
    listener: (error: Error) => void | Promise<void>,
  ) => this;
}
```

After auto-fixing:

```typescript
interface MyEventEmitter extends EventEmitter {
  on: (event: 'my-event-1', listener: () => void | Promise<void>) => this;
  on: (
    event: 'error',
    listener: (error: Error) => void | Promise<void>,
  ) => this;
}
```

Which results in a TS compiler error `Duplicate identifier 'on'.`.

**Expected Result**

Either the syntax is not reported as incorrect or generated code is compilable.

**Actual Result**

A linting error is reported, but the generated, auto-fixed code is not compilable.

**Additional Info**

Not relevant.

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `2.28.0` |
| `@typescript-eslint/parser`        | `2.28.0` |
| `TypeScript`                       | `3.8.3` |
| `ESLint`                           | `6.8.0` |
| `node`                             | `12.16.1` |
| `npm`                              | `6.13.4` |

**Comment by bradzacher** [comment]
The correct fix for overloads should be

```ts
interface Foo {
  method(): one
  method(): two
}

interface Foo {
  method: (() => one) & (() => two)
}
```

Happy to accept a PR to fix this
